nomad/plan_endpoint.go

```diff
@@ -20,11 +20,22 @@ func (p *Plan) Submit(args *structs.PlanRequest, reply *structs.PlanResponse) er
 	defer metrics.MeasureSince([]string{"nomad", "plan", "submit"}, time.Now())
 
 	// Submit the plan to the queue
-	future, err := p.srv.planQueue.Enqueue(args.Plan)
+	plan := args.Plan
+	future, err := p.srv.planQueue.Enqueue(plan)
 	if err != nil {
 		return err
 	}
 
+	// Pause the Nack timer for the eval as it is making progress as long as it
+	// is in the plan queue. We resume immediately after we get a result to
+	// handle the case that the receiving worker dies.
+	id := plan.EvalID
+	token := plan.EvalToken
+	if err := p.srv.evalBroker.PauseNackTimeout(id, token); err != nil {
```

Comment:
Can we move this before the Enqueue? That way we can avoid Enqueue if its not outstanding or there is a token mismatch